### DISCUSSION TITLE
getSubscription on IOS don't work

### DISCUSSION THREAD

**Discussion by Antonin-Thobois** [hyochan/react-native-iap#2166] [Q&A]
Hello, I’ve been using react-native-iap for 2 months now and I have a problem on IOS to use the getSubscription function. On Android it works perfectly.

I have on App store Connect, a subcription group with 2 subscriptions (1 month and 1 year). I used all the IDs I could find on the App store Connect but the getSubscription function always returns an empty array.

Can someone help me ? Please.

**Comment by ajouve** [comment]
Hi,

Did you configure the sandbox account on your phone ?

**Reply by Antonin-Thobois** [reply]
Hi,

Yes I have already configured the sandbox account but it always returns an empty array.

I also tried to put the application into prod but nothing happens too and the function always returns me an empty array.

Thank you for your answer.

**Reply by CoreyBovalina** [reply]
ever figure this out?  I'm trying to get this to work on iOS now, I setup a sandbox, and initConnection returns true, but when I call getSubscriptions or requestSubscription with my sky, I either get an empty array or it says product id not found.

**Comment by andresesfm** [comment]
Did you try Sk2 mode? I'm thinking that maybe that kind of subscription is not supported on Sk1
